src/mapml/layers/FeatureLayer.js

```diff
@@ -144,26 +143,38 @@ export var FeatureLayer = L.FeatureGroup.extend({
       }
     },
 
-    _resetFeatures : function (zoom){
+    _resetFeatures : function () {
       this.clearLayers();
       // since features are removed and re-added by zoom level, need to clean the feature index before re-adding
       if(this._map) this._map.featureIndex.cleanIndex();
-      if(this._features && this._features[zoom]){
-        for(let k =0;k < this._features[zoom].length;k++){
-          this.addLayer(this._features[zoom][k]);
+      let map = this._map || this.options._leafletLayer._map;
+      if(this._features) {
+        for (let zoom in this._features) {
+          for(let k =0;k < this._features[zoom].length;k++){
+            let feature = this._features[zoom][k],
+                checkRender = feature._checkRender(map.getZoom(), this.zoomBounds.minZoom, this.zoomBounds.maxZoom);
+            if (checkRender) {
+              this.addLayer(feature);
+            }
+          }
         }
       }
     },
 
+    // substituted with the _checkRender function in FeatureGroup.js
     _clampZoom : function(zoom){
```

Comment:
The code is no longer needed for the latest changes. I have kept it there and plan to remove it when the PR is ready to go, just to prevent any possible side effects. Alternatively, I could comment it out now as a more cautious approach. What do you think would be the best course of action?

Reply:
I think it would be good to finalize it now, just so there are no side effects we may encounter when we are merging! Always good to remove unnecessary code!